### PR TITLE
For #6639: The new onboarding is wronlgy displayed and scrollable in …

### PR DESCRIPTION
**Pull request by indurs**
…RTL languages

File modified

OnboardingFragment

* The parent layout(ConstraintLayout) of KeyFeatureCard composable is replaced with Row

* Onboarding description is aligned center horizontally

Onboarding screen for LTR and RTL languages

Screenshot_20220315-064324  Screenshot_۲۰۲۲۰۳۱۵-۱۰۰۶۲۳

Co-Authored-By: Mugurell <Mugurell@users.noreply.github.com>

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

**Comment by ionutbedregeanu** [comment]
The text doesn't look aligned properly(related to the icon) as in Figma design

**Comment by indurs** [comment]
@ionutbedregeanu , do I need to make changes?

**Comment by ionutbedregeanu** [comment]
> @ionutbedregeanu , do I need to make changes?

In the attached image from my comment the title of every sub-section seems to be centered vertically related to its icon.
In your attached screenshots seems to be a vertical offset between between the text and icon

**Comment by indurs** [comment]
@ionutbedregeanu , I'll look into that. I couldn't see the figma design. I'm seeing this error from Figma. 

File not found

Either this file doesn’t exist or you don’t have permission to view it. Ask the file owner to verify the link and/or update permissions

**Comment by indurs** [comment]
Sure. No problem..

**Comment by indurs** [comment]
@ionutbedregeanu, I updated the alignment for the icon and text. Could you please verify whether it's fine? Thank You

Screenshot_۲۰۲۲۰۳۱۷-۱۰۴۳۰۷  Screenshot_20220317-104202

File: app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt [hunk 1]
```diff
@@ -149,34 +141,24 @@ class OnboardingFragment : Fragment() {
...
-                modifier = Modifier.constrainAs(description) {
-                    top.linkTo(title.bottom)
-                    start.linkTo(title.start)
-                    end.linkTo(parent.end)
-                    width = Dimension.fillToConstraints
-                },
-                style = focusTypography.onboardingDescription
-            )
+        Row(modifier = Modifier.fillMaxWidth().padding(start=16.dp)){
+            Column() {
```
Comment (ionutbedregeanu) [inline app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:145]: On a first look I have mixed feelings about having a column for only one child. I will take a second look to see what we can do about that

Comment (indurs) [reply app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:145]: Sure, we will look into that

Comment (indurs) [reply app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:145]: @ionutbedregeanu , I think it's not necessary as you mentioned. We can remove the column.

```
@Composable
    fun KeyFeatureCard(iconId: Int, titleId: Int, descriptionId: Int) {
        Row(modifier = Modifier.fillMaxWidth().padding(start=16.dp)){
            Image(
                painter = painterResource(iconId),
                colorFilter = ColorFilter.tint(color = focusColors.onboardingKeyFeatureImageTint),
                contentDescription = getString(R.string.app_name),
            )

            Column(modifier = Modifier.padding(start = 16.dp)) {
                Text(
                    text = stringResource(id = titleId),
                    style = focusTypography.onboardingSubtitle
                )
                Text(
                    text = stringResource(id = descriptionId, getString(R.string.onboarding_short_app_name)),
                    style = focusTypography.onboardingDescription
                )
            }
        }
    }
```

Comment (indurs) [reply app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:145]: @ionutbedregeanu , I also added top 4.dp padding for the first text right of the icon

```
@Composable
    fun KeyFeatureCard(iconId: Int, titleId: Int, descriptionId: Int) {
        Row(modifier = Modifier.fillMaxWidth().padding(start=16.dp)){
            Image(
                painter = painterResource(iconId),
                colorFilter = ColorFilter.tint(color = focusColors.onboardingKeyFeatureImageTint),
                contentDescription = getString(R.string.app_name),
            )

            Column(modifier = Modifier.padding(start = 16.dp, top = 4.dp)) {
                Text(
                    text = stringResource(id = titleId),
                    style = focusTypography.onboardingSubtitle
                )
                Text(
                    text = stringResource(id = descriptionId, getString(R.string.onboarding_short_app_name)),
                    style = focusTypography.onboardingDescription
                )
            }
        }
    }
```

File: app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt [hunk 2]
```diff
@@ -149,34 +141,24 @@ class OnboardingFragment : Fragment() {
 
     @Composable
     fun KeyFeatureCard(iconId: Int, titleId: Int, descriptionId: Int) {
-        ConstraintLayout(modifier = Modifier.fillMaxWidth()) {
-            val (image, title, description) = createRefs()
-            Image(
-                painter = painterResource(iconId),
-                colorFilter = ColorFilter.tint(color = focusColors.onboardingKeyFeatureImageTint),
-                contentDescription = getString(R.string.app_name),
-                modifier = Modifier.constrainAs(image) { top.linkTo(parent.top) }
```
Comment (ionutbedregeanu) [inline app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:158]: To fix the RTL bug we can only constraint the image to start where the parent does start.linkTo(parent.start)

Screenshot 2022-03-17 at 10 49 39

Comment (ionutbedregeanu) [reply app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:158]: In this way we don't need to replace the constraint layout with a row and add paddings

Comment (ionutbedregeanu) [reply app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:158]: yes

Comment (ionutbedregeanu) [reply app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:158]: Can you please squash the commits?

Comment (indurs) [reply app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:158]: I squashed the commits

File: app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt [hunk 3]
```diff
@@ -12,15 +12,7 @@ import android.view.View
...
-import androidx.compose.foundation.layout.Arrangement
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.fillMaxHeight
-import androidx.compose.foundation.layout.fillMaxWidth
-import androidx.compose.foundation.layout.height
-import androidx.compose.foundation.layout.padding
-import androidx.compose.foundation.layout.size
-import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.*
```
Comment (ionutbedregeanu) [inline app/src/main/java/org/mozilla/focus/fragment/onboarding/OnboardingFragment.kt:15]: Sorry, @indurs but now I noticed this * import. Can you please revert this and after that squash the commits? It's not a good practice to use * import. It's the last change I ask. Thanks!

**Review by ionutbedregeanu — APPROVED** [review]

**Comment by ionutbedregeanu** [comment]
I'm closing this pr in favor of https://github.com/mozilla-mobile/focus-android/pull/6689

**Pull request closed without merging**